### PR TITLE
reverse integer problem solved

### PR DESCRIPTION
**Pull request by unicode368**

### CODE REVIEW

File: src/ReverseInteger.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+public class ReverseInteger {
+    public static int reverse(int x) {
+        long res = 0;
+        int mod = Math.abs(x);
+        int i = (int) Math.log10(mod);
+        if(i == 0 && (mod > 10 || mod < -10)) {
```
Comment (ybystrov) [inline src/ReverseInteger.java]: What is the intention here? (mod < -10) can never be true because mod is a positive number

Comment (ybystrov) [inline src/ReverseInteger.java]: I do not understand the purpose of this check. Can you give an example of when it is needed?

File: src/ReverseInteger.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            return 0;
+        }
+        while(x != 0 && i >= 0) {
+            res += (x % 10) * Math.pow(10, i);
+            x /= 10;
+            i--;
+        }
+        if(res > 2147483647 ||
```
Comment (ybystrov) [inline src/ReverseInteger.java]: Use Integer.MAX_VALUE and Integer.MIN_VALUE constants

File: test/ReverseIntegerTest.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                { 123 , 321, "Plain reverse" },
+                { -123 , -321, "Reverse negative number" },
+                { 9 , 9, "Less than 10" },
+                { 0 , 0, "Reverse zero" },
+                {-2147483648, 0, "Reversed number out of bounds"}
+        });
+    }
+
+    @Test
+    public void testAddTwoNumbers() {
```
Comment (ybystrov) [inline test/ReverseIntegerTest.java]: Why did you call the test method testAddTwoNumbers?

File: src/ReverseInteger.java (hunk 1 above)
Comment (unicode368) [reply src/ReverseInteger.java]: image
Integer.MIN = -2147483648. If you apply Math.abs to this number - the minus will remain as Integer.MIN is negation of itself, thus mod can be a negative number. Although the condition can be simplified indeed to i == 0 && mod == Integer.MIN_VALUE

File: src/ReverseInteger.java (hunk 2 above)
Comment (unicode368) [reply src/ReverseInteger.java]: Okay

File: test/ReverseIntegerTest.java (hunk 3 above)
Comment (unicode368) [reply test/ReverseIntegerTest.java]: Now the test method has name testHasCycle and name of the test itself is defined by input parameters, provided by provideParameters() method

**Pull request merged**